### ISSUE TITLE
Implicit type coersion of function to boolean in ternary operator

### ISSUE THREAD

**Issue by kenrick95** [microsoft/TypeScript#36921]
**TypeScript Version:**  3.7.x-dev.20200220

**Search Terms:** Implicit type coersion function ternary

**Code**

```ts
function isSomething(): boolean {
  return false
}
const haha = isSomething ? 'a' : 'b';
```

**Expected behavior:** Typescript should throw error in `= isSomething ?` instead of coercing `isSomething` to boolean

**Actual behavior:** Typescript coerced `isSomething` (function) to boolean

**Playground Link:**  [link](http://www.typescriptlang.org/play/?ts=3.9.0-dev.20200220#code/GYVwdgxgLglg9mABDAzgZTgWwKZQBYxgDmAFAJQBciARnHADbYCGSA3gFCKIBOuI3SYE3ops7AL7sICFFER4mCxAF5k6LLgLFEAfkQByJvsRV91fQG52QA)

**Related Issues:** 

- [Flow #5674](https://github.com/facebook/flow/issues/5674)

**Comment by IllusionMH** [comment]
Fixed in https://github.com/microsoft/TypeScript/pull/36402 (will be in 3.9)

**Comment by kenrick95** [comment]
Omg thanks a lot @IllusionMH! Apologies for this duplicate issue.
